packages/block-library/src/social-link/index.php

```diff
@@ -92,6 +92,24 @@ function register_block_core_social_link() {
 			)
 		);
 	}
+
```

Comment:
I'd love to get rid of the loop above right away, in which all `core/social-link-<foo>` types are generated, but in order to do that we need a better plan for back-compat.

Right now only the editor has a back-compat provision via the change in `packages/blocks/src/api/parser.js`. How could we best handle this for server-side block rendering for the front-end?

Reply:
You should check the HTML output of the edit post page. All blocks registered here will have a huge impact on the size of the page. In particular, the part that registers blocks to work with the `ServerSideRender` component (which is also pointless in this case as they will never be consumed).

At the same time, it's something that was never part of the WordPress core so we can provide the backward-compatibility only for the Gutenberg plugin and call it the day 😄

Reply:
> At the same time, it's something that was never part of the WordPress core so we can provide the backward-compatibility only for the Gutenberg plugin and call it the day 😄

Yeah, that seems like the right way to go. Let's address in a subsequent PR.

Reply:
#20098